CMakeLists.txt

```diff
@@ -20,7 +20,14 @@ option(REQUIRE_IFI "Abort if libIFI is not found ; enables BUILD_WITH_IFI=ON" OF
 option(BUILD_WITH_GTG   "Build NCEPpost with NCAR/GTG" OFF)
 option(ENABLE_DOCS "Enable generation of doxygen-based documentation." OFF)
 
-if(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo|MinSizeRel)$")
+# Set the build type.
+if(DEFINED CMAKE_BUILD_TYPE)
+  string(TOUPPER ${CMAKE_BUILD_TYPE} UPPER_CMAKE_BUILD_TYPE)
+else()
+  set(UPPER_CMAKE_BUILD_TYPE "")
+endif()
+
+if(NOT UPPER_CMAKE_BUILD_TYPE MATCHES "^(DEBUG|RELEASE|RELWITHDEBINFO|MINSIZEREL)$")
```

Comment:
`CMAKE_BUILD_TYPE` should not be changed in this manner.
See [here](https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html)

Reply:
Not sure why this is necessary.  It seems to rectify the incorrect call in the cmake invocation.